blockchain/blockchain.go

```diff
@@ -1976,3 +1977,19 @@ func (bc *BlockChain) GetNonceInCache(addr common.Address) (uint64, bool) {
 	}
 	return 0, false
 }
+
+// CheckBlockChainVersion checks the version of the current database and upgrade if possible.
+func CheckBlockChainVersion(chainDB database.DBManager) error {
+	bcVersion := chainDB.ReadDatabaseVersion()
+	if bcVersion != nil && *bcVersion > BlockChainVersion {
+		return fmt.Errorf("database version is v%d, Klaytn %s only supports v%d", *bcVersion, params.Version, BlockChainVersion)
+	} else if bcVersion == nil || *bcVersion < BlockChainVersion {
+		bcVersionStr := "N/A"
+		if bcVersion != nil {
+			bcVersionStr = strconv.Itoa(int(*bcVersion))
+		}
+		logger.Warn("Upgrade database version", "from", bcVersionStr, "to", BlockChainVersion)
+		chainDB.WriteDatabaseVersion(BlockChainVersion)
+	}
+	return nil
```

Comment:
`chainDB.WriteDatabaseVersion(blockchain.BlockChainVersion)` is not necessary for this case?
```suggestion
	chainDB.WriteDatabaseVersion(blockchain.BlockChainVersion)
	return nil
```

Reply:
@KimKyungup No, it is not needed. Because if the code reaches there, it means that the node already has DatabaseVersion equal to blockchain.BlockChainVersion. Hence, no need to overwrite it with same value.